### PR TITLE
[MRG+1] Added plot_image method in epoch, calling plot_epochs_image from viz

### LINKED ISSUE #2809: Epochs should have plot_image as method
Currently, you have to use `mne.viz.plot_epochs_image` for plotting. It would be convenient to have it as a method like the other plotting functions.

### PR DESCRIPTION
**Pull request by kaichogami**
This closes #2809. I did not add tests as they are already taken care at `viz.tests`. 
@teonlamont @jona-sassenhagen please review when you have time.

### CODE REVIEW

File: mne/epochs.py [hunk 1]
```diff
@@ -1084,6 +1085,65 @@ def plot_drop_log(self, threshold=0, n_max_plot=20, subject='Unknown',
...
+        overlay_times : array-like, shape (n_epochs,) | None
+            If not None the parameter is interpreted as time instants in
+            seconds and is added to the image. It is typically useful to
+            display reaction times. Note that it is defined with respect
+            to the order of epochs such that overlay_times[0] corresponds
+            to epochs[0].
+        Returns
+        -------
+        figs : the list of matplotlib figures
+            One figure per channel displayed
```
Comment (kingjr) [inline mne/epochs.py]: displayed.

File: mne/epochs.py [hunk 2]
```diff
@@ -1084,6 +1085,65 @@ def plot_drop_log(self, threshold=0, n_max_plot=20, subject='Unknown',
...
+            None a new figure is created. Defaults to None.
+        overlay_times : array-like, shape (n_epochs,) | None
+            If not None the parameter is interpreted as time instants in
+            seconds and is added to the image. It is typically useful to
+            display reaction times. Note that it is defined with respect
+            to the order of epochs such that overlay_times[0] corresponds
+            to epochs[0].
+        Returns
+        -------
+        figs : the list of matplotlib figures
```
Comment (kingjr) [inline mne/epochs.py]: figs: list of matplotlib figures

File: mne/epochs.py [hunk 3]
```diff
@@ -1084,6 +1085,65 @@ def plot_drop_log(self, threshold=0, n_max_plot=20, subject='Unknown',
...
+            Show figure if True.
+        units : dict | None
+            The units of the channel types used for axes lables. If None,
+            defaults to `units=dict(eeg='uV', grad='fT/cm', mag='fT')`.
+        scalings : dict | None
+            The scalings of the channel types to be applied for plotting.
+            If None, defaults to `scalings=dict(eeg=1e6, grad=1e13, mag=1e15,
+            eog=1e6)`
+        cmap : matplotlib colormap
+            Colormap.
```
Comment (kingjr) [inline mne/epochs.py]: Defaults to ?

File: mne/epochs.py [hunk 4]
```diff
@@ -1084,6 +1085,65 @@ def plot_drop_log(self, threshold=0, n_max_plot=20, subject='Unknown',
...
+            (data.shape[1] == len(times)
+        show : bool
+            Show figure if True.
+        units : dict | None
+            The units of the channel types used for axes lables. If None,
+            defaults to `units=dict(eeg='uV', grad='fT/cm', mag='fT')`.
+        scalings : dict | None
+            The scalings of the channel types to be applied for plotting.
+            If None, defaults to `scalings=dict(eeg=1e6, grad=1e13, mag=1e15,
+            eog=1e6)`
```
Comment (kingjr) [inline mne/epochs.py]: 1e6)'.

File: mne/epochs.py [hunk 5]
```diff
@@ -1084,6 +1085,65 @@ def plot_drop_log(self, threshold=0, n_max_plot=20, subject='Unknown',
...
+            The max value in the image. The unit is uV for EEG channels,
+            fT for magnetometers and fT/cm for gradiometers
+        colorbar : bool
+            Display or not a colorbar
+        order : None | array of int | callable
+            If not None, order is used to reorder the epochs on the y-axis
+            of the image. If it's an array of int it should be of length
+            the number of good epochs. If it's a callable the arguments
+            passed are the times vector and the data as 2d array
+            (data.shape[1] == len(times)
```
Comment (kingjr) [inline mne/epochs.py]: len(times).

File: mne/epochs.py [hunk 6]
```diff
@@ -1084,6 +1085,65 @@ def plot_drop_log(self, threshold=0, n_max_plot=20, subject='Unknown',
...
+            the epoch axis to apply in the image. If 0., no smoothing is
+            applied.
+        vmin : float
+            The min value in the image. The unit is uV for EEG channels,
+            fT for magnetometers and fT/cm for gradiometers
+        vmax : float
+            The max value in the image. The unit is uV for EEG channels,
+            fT for magnetometers and fT/cm for gradiometers
+        colorbar : bool
+            Display or not a colorbar
```
Comment (kingjr) [inline mne/epochs.py]: colorbar.

File: mne/epochs.py [hunk 7]
```diff
@@ -1084,6 +1085,65 @@ def plot_drop_log(self, threshold=0, n_max_plot=20, subject='Unknown',
...
+        picks : int | array-like of int | None
+            The indices of the channels to consider. If None, the first
+            five good channels are plotted.
+        sigma : float
+            The standard deviation of the Gaussian smoothing to apply along
+            the epoch axis to apply in the image. If 0., no smoothing is
+            applied.
+        vmin : float
+            The min value in the image. The unit is uV for EEG channels,
+            fT for magnetometers and fT/cm for gradiometers
```
Comment (kingjr) [inline mne/epochs.py]: gradiometers. Don't forget the final dot. Same everywhere else.

File: mne/epochs.py [hunk 8]
```diff
@@ -1084,6 +1085,65 @@ def plot_drop_log(self, threshold=0, n_max_plot=20, subject='Unknown',
...
+            to epochs[0].
+        Returns
+        -------
+        figs : the list of matplotlib figures
+            One figure per channel displayed
+        """
+        return plot_epochs_image(self, picks, sigma, vmin, vmax, colorbar,
+                                 order, show, units, scalings, cmap, fig,
+                                 overlay_times)
+
```
Comment (kingjr) [inline mne/epochs.py]: Here I would explicit all kwargs to avoid any future error.

File: mne/epochs.py [hunk 9]
```diff
@@ -1084,6 +1085,65 @@ def plot_drop_log(self, threshold=0, n_max_plot=20, subject='Unknown',
                              color=color, width=width, ignore=ignore,
                              show=show)
 
+    def plot_image(self, picks=None, sigma=0., vmin=None,
```
Comment (teonlamont) [inline mne/epochs.py]: is this the same docstring for the `plot_epochs_image`, just minus the `epochs` parameter?

**Comment by agramfort** [comment]
for the test update the tests so it does epochs.plot_image

by doing so the coverage will not go down

**Comment by jona-sassenhagen** [comment]
Can you propagate any changes to `mne.viz.plot_epochs_images`?

File: mne/epochs.py (hunk 3 above)
Comment (kaichogami) [inline mne/epochs.py]: Defaults to this [line](https://github.com/mne-tools/mne-python/blob/master/mne/defaults.py#L17).

File: mne/epochs.py (hunk 9 above)
Comment (kaichogami) [inline mne/epochs.py]: Yes, it the same docstring from `plot_epochs_image`.

**Comment by kaichogami** [comment]
@jona-sassenhagen do you mean to update the docstring changes made here to `mne.viz.plot_epochs_images` in a separate PR?
@agramfort I simply made a call to `epochs.plot_image`, without really testing anything. Is that what should be done?

**Comment by jona-sassenhagen** [comment]
>@jona-sassenhagen do you mean to update the docstring changes made here to mne.viz.plot_epochs_images in a separate PR?

No, in this.

>@agramfort I simply made a call to epochs.plot_image, without really testing anything. Is that what should be done?

No, you should make a dedicated test - look up where mne.viz.plot_epochs_images is tested, and either convert one of the tests to using the method, or add a new test. This is important for coverage.

**Comment by dengemann** [comment]
Looks good to me!

**Comment by agramfort** [comment]
@kaichogami you should update all the tests that use plot_epochs_images and call the epochs.plot_image function. It avoids wasting test time due to addition of redundant tests.

**Comment by kaichogami** [comment]
@agramfort I changed the tests as you suggested. Let me know if its alright now.

**Comment by kaichogami** [comment]
I am not sure why tests are failing. I can see the issue being related to doc formatting, although I have no idea where. Any help would be appreciated. 
Thank you!

**Comment by agramfort** [comment]
perfect. thanks @kaichogami

**Comment by kaichogami** [comment]
@agramfort Why was it merged when the tests are failing?

**Comment by dengemann** [comment]
> @agramfort Why was it merged when the tests are failing?

@kaichogami sometimes the tests fail because travis has some unrelated issues.

**Comment by kaichogami** [comment]
@dengemann Alright. Thank you for letting me know! :)

**Pull request merged**